packages/calcite-components/src/components/dropdown/dropdown.tsx

```diff
@@ -634,29 +636,34 @@ export class Dropdown
     }
   }
 
-  private focusOnFirstActiveOrFirstItem = (): void => {
-    this.getFocusableElement(
-      this.getTraversableItems().find((item) => item.selected) || this.items[0],
-    );
+  private focusOnFirstActiveOrDefaultItem = (focusLastItem: boolean): void => {
+    const selectedItem = this.getTraversableItems().find((item) => item.selected);
+    if (selectedItem) {
+      this.focusDropdownItemElement(selectedItem);
+      return;
+    } else {
+      this.focusDropdownItemElement(
+        focusLastItem ? this.items[this.items.length - 1] : this.items[0],
+      );
+    }
   };
 
-  private getFocusableElement(item: HTMLCalciteDropdownItemElement): void {
+  private focusDropdownItemElement(item: HTMLCalciteDropdownItemElement): void {
     if (!item) {
       return;
     }
-
     focusElement(item);
   }
 
-  private toggleOpenEnd = (): void => {
-    this.focusOnFirstActiveOrFirstItem();
-    this.el.removeEventListener("calciteDropdownOpen", this.toggleOpenEnd);
+  private toggleOpenEnd = (focusLastItem: boolean): void => {
+    this.focusOnFirstActiveOrDefaultItem(focusLastItem);
+    this.el.removeEventListener("calciteDropdownOpen", () => this.toggleOpenEnd(focusLastItem));
```

Comment:
Sidebar: I think we could remove the explicit event listener add/remove by leveraging the [`OpenCloseComponent` hooks](https://github.com/Esri/calcite-design-system/blob/1cb3d9055f0f4d36effd83c1543e1baae78c275f/packages/calcite-components/src/components/dropdown/dropdown.tsx#L538-L552).

Reply:
I think we may have to depend on explicit event listener to avoid focusing items when `open` attribute is changed programmatically. 

```
  onOpen(): void {
    this.calciteDropdownOpen.emit();
    this.focusOnFirstActiveOrDefaultItem();
  }
  ```
  
  Tried the above 👆 and the issue with that is when `open` is toggled without user interaction (programmatically) , `onOpen( )` method is triggered and focus shift happens. Is this expected?

Reply:
Took a closer look after our conversation yesterday and yes, this is expected. Our open/close events emit for both programmatic and user interaction (https://github.com/Esri/calcite-design-system/pull/7613/files has a draft for guidelines on this). 

Does this impact any dropdown workflows? If not, let's proceed. Otherwise, we can restore the explicit event listeners and refactor to use the open/close events in the next release.